packages/core/src/embeddings/types.ts

```diff
@@ -10,15 +10,60 @@ export enum SimilarityType {
   EUCLIDEAN = "euclidean",
 }
 
+export type Embedding = number[];
+
+const DEFAULT_EMBED_BATCH_SIZE = 10;
+
+/**
+ * Base embedding class
+ */
 export abstract class BaseEmbedding {
+  embedBatchSize = DEFAULT_EMBED_BATCH_SIZE;
+
   similarity(
-    embedding1: number[],
-    embedding2: number[],
+    embedding1: Embedding,
+    embedding2: Embedding,
     mode: SimilarityType = SimilarityType.DEFAULT,
   ): number {
     return similarity(embedding1, embedding2, mode);
   }
 
-  abstract getTextEmbedding(text: string): Promise<number[]>;
-  abstract getQueryEmbedding(query: string): Promise<number[]>;
+  abstract getTextEmbedding(textS: string | string[]): Promise<Embedding>;
```

Comment:
@EmanuelCampos I agree to move forward with what we have. We don't want this PR to go stale :) 

To merge this PR, I suggest the following:

1. Use the same contract as in Python (will be less confusing for our users):
```
getTextEmbedding(text: string): Promise<Embedding>; 
getQueryEmbedding(query: string): Promise<Embedding>; 
getTextEmbeddings(texts: string[]): Promise<Embedding>; 
getTextEmbeddingsBatch(texts: string[]): Promise<Array<Embedding>>; 
```
Like in Python, the default implementation of `getTextEmbeddings` just iterates over all strings and calls `getTextEmbedding` - subclasses can override this behavior and send multiple strings in one request (like your OpenAI version).  

2. Either use `Embedding` type consistently for all embeddings, or remove introducing this type (It's a good idea, but it should be everywhere if we do it)